Content.Server/Medical/CryoPodSystem.cs

```diff
@@ -276,10 +276,17 @@ private void OnGasAnalyzed(Entity<CryoPodComponent> entity, ref GasAnalyzerScanE
         if (!TryComp(entity, out CryoPodAirComponent? cryoPodAir))
             return;
 
-        args.GasMixtures ??= new Dictionary<string, GasMixture?> { { Name(entity.Owner), cryoPodAir.Air } };
+        args.GasMixtures ??= new List<(string, GasMixture?)>();
+        args.GasMixtures.Add((Name(entity.Owner), cryoPodAir.Air));
         // If it's connected to a port, include the port side
-        if (_nodeContainer.TryGetNode(entity.Owner, entity.Comp.PortName, out PipeNode? port))
-            args.GasMixtures.Add(entity.Comp.PortName, port.Air);
+        // multiply by volume fraction to make sure to send only the gas inside the analyzed pipe element, not the whole pipe system
+        if (_nodeContainer.TryGetNode(entity.Owner, entity.Comp.PortName, out PipeNode? port) && port.Air.Volume != 0f)
+        {
+            var portAirLocal = port.Air.Clone();
+            portAirLocal.Multiply(port.Volume / port.Air.Volume);
+            portAirLocal.Volume = port.Volume;
+            args.GasMixtures.Add((entity.Comp.PortName, portAirLocal));
```

Comment:
I could add a function for this in [GasMixture.cs](https://github.com/space-wizards/space-station-14/blob/master/Content.Server/Atmos/GasMixture.cs). Although the existing functions in there are confusingly named. One could think Multiply(...), RemoveVolume(...), or RemoveRatio(...) would do what is needed here, but these do not adjust the volume. Also we need a clone since we do not want to modify the existing GasMixture in the pipe. Do you have a name suggestion for the new function which would make it clear that this one would work different from the others? MultiplyToVolume would be possible, but easy to confuse with Multiply(...) which modifies an existing GasMixture instead of returning a copy.

Reply:
Like you pointed out, we can get this in and look at this later. Thanks for looking at the options.